src/java/src/com/facebook/csslayout/CSSNode.java

```diff
@@ -400,4 +400,20 @@ public void setDefaultPadding(int spacingType, float padding) {
       dirty();
     }
   }
+
+  /**
+   * Resets this instance to its default state. This method is meant to be used when
+   * recycling {@link CSSNode} instances.
+   */
+  public void reset() {
+    style.reset();
+    layout.reset();
+    lineIndex = 0;
+    mLayoutState = LayoutState.DIRTY;
+    mMeasureFunction = null;
```

Comment:
My use case is for a system that recycles nodes across different layout trees. This requires the reset() call to restore the CSSNode to its original state.